src/control/build/interop.go

```diff
@@ -0,0 +1,165 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package build
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// MaxMajorDelta is the default maximum major version delta
+	// allowed between two components.
+	MaxMajorDelta = 0
+	// MaxMinorDelta is the default maximum minor version delta
+	// allowed between two components.
+	MaxMinorDelta = 2
+)
+
+// Component is a component of the system.
+type Component string
+
+// Matches returns true if the component is equal to other
+// or one of the components is a wildcard.
+func (c Component) Matches(other Component) bool {
+	return c == other || c == ComponentAny || other == ComponentAny
+}
+
+func (c Component) String() string {
+	return string(c)
+}
+
+var (
+	// ComponentAny is a wildcard component.
+	ComponentAny = Component("")
+	// ComponentServer represents the control plane server.
+	ComponentServer = Component("server")
+	// ComponentAdmin represents the Control API client.
+	ComponentAdmin = Component("admin")
+	// ComponentAgent represents the compute node agent.
+	ComponentAgent = Component("agent")
+)
+
+// NewVersionedComponent creates a new VersionedComponent.
+func NewVersionedComponent(comp Component, version string) (*VersionedComponent, error) {
+	v, err := NewVersion(version)
+	if err != nil {
+		return nil, err
+	}
+
+	switch comp {
+	case ComponentServer, ComponentAdmin, ComponentAgent, ComponentAny:
+		return &VersionedComponent{
+			Component: comp,
+			Version:   *v,
+		}, nil
+	default:
+		return nil, errors.Errorf("invalid component %q", comp)
+	}
+}
+
+// VersionedComponent is a component with a version.
+type VersionedComponent struct {
+	Version   Version
+	Component Component
+}
+
+func (vc *VersionedComponent) String() string {
+	return fmt.Sprintf("%s:%s", vc.Component, vc.Version)
```

Comment:
I am probably missing something with golang, but it seems to be prettier like this ?
```suggestion
	return fmt.Sprintf("%s:%s", vc.Component, vc.Version.String())
```

Reply:
Can you explain what you mean by "prettier"? Do you mean the code or the output? Currently, the stringer results in output like this: `incompatible components: server:1.0.0, server:1.1.0`

Reply:
Not black magic exactly, just a format rule: https://pkg.go.dev/fmt#Stringer